common/prque/sstack.go

```diff
@@ -81,7 +115,38 @@ func (s *sstack) Len() int {
 // Compares the priority of two elements of the stack (higher is first).
 // Required by sort.Interface.
 func (s *sstack) Less(i, j int) bool {
-	return (s.blocks[i/blockSize][i%blockSize].priority - s.blocks[j/blockSize][j%blockSize].priority) > 0
+	var result bool
+
+	switch s.priorityType {
+	case Int:
+		i := (s.blocks[i/blockSize][i%blockSize].priority).(int)
+		j := (s.blocks[j/blockSize][j%blockSize].priority).(int)
```

Comment:
What about change those i,j variables to different name?

Reply:
It is reflected in #840 PTAL :)